### PR TITLE
Add ratings to articles

### PR DESCRIPTION
**Pull request by AmberWilkie**
https://www.pivotaltracker.com/story/show/130181931

Display top-rated articles on home page.

### CODE REVIEW

File: app/models/average_cache.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+class AverageCache < ActiveRecord::Base
+  belongs_to :rater, :class_name => "User"
```
Comment (tochman) [inline app/models/average_cache.rb]: Hash syntax `class_name: "User"`

File: app/models/average_cache.rb [hunk 2]
```diff
@@ -0,0 +1,4 @@
...
+  belongs_to :rateable, :polymorphic => true
```
Comment (tochman) [inline app/models/average_cache.rb]: Same here...

File: app/models/overall_average.rb [hunk 3]
```diff
@@ -0,0 +1,4 @@
+class OverallAverage < ActiveRecord::Base
+  belongs_to :rateable, :polymorphic => true
```
Comment (tochman) [inline app/models/overall_average.rb]: Same here. Hash syntax

File: app/models/rate.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+class Rate < ActiveRecord::Base
+  belongs_to :rater, class_name: "User"
+  belongs_to :rateable, polymorphic: true
+
+  #attr_accessible :rate, :dimension
```
Comment (tochman) [inline app/models/rate.rb]: ??

File: app/models/rating_cache.rb [hunk 5]
```diff
@@ -0,0 +1,3 @@
+class RatingCache < ActiveRecord::Base
+  belongs_to :cacheable, :polymorphic => true
```
Comment (tochman) [inline app/models/rating_cache.rb]: Same here. Hash syntax.

File: config/routes.rb [hunk 6]
```diff
@@ -1,4 +1,5 @@
 Rails.application.routes.draw do
+  post '/rate' => 'rater#create', :as => 'rate'
```
Comment (tochman) [inline config/routes.rb]:
```
post '/rate', controller: :rate, action: :create, as: 'rate'
```

**Comment by tochman** [comment]
I would at least add some regression specs for all the models generated by the gem.

File: features/ratings.feature [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    | title   | body             |
+    | One     | One awesome text |
+    | Two     | Two awesome text |
+    | Three   | Three good texts |
+
+Scenario: Amber rates Hanna's articles
+  Given I am logged in as "amber@random.com"
+  And I am on the article page for article "One"
+  Then I should see "Rate this article"
+  # Testing this further would require all kinds of Javascript
```
Comment (tochman) [inline features/ratings.feature]: is it possible to click on a star? Directly using Capybara or by `execute_script`?

File: app/models/average_cache.rb (hunk 1 above)
Comment (AmberWilkie) [reply app/models/average_cache.rb]: done

File: app/models/average_cache.rb (hunk 2 above)
Comment (AmberWilkie) [reply app/models/average_cache.rb]: done. These were all auto-generated by ratyrate and I honestly couldn't figure out what those DB associations looked like.

File: app/models/overall_average.rb (hunk 3 above)
Comment (AmberWilkie) [reply app/models/overall_average.rb]: done

File: app/models/rate.rb (hunk 4 above)
Comment (AmberWilkie) [reply app/models/rate.rb]: No idea. Deleted it.

File: app/models/rating_cache.rb (hunk 5 above)
Comment (AmberWilkie) [reply app/models/rating_cache.rb]: done

File: config/routes.rb (hunk 6 above)
Comment (AmberWilkie) [reply config/routes.rb]: done

**Pull request closed without merging**